src/link/reference/source.rs

```diff
@@ -96,7 +97,13 @@ impl ReferenceSource {
         }
 
         let group = config::generate_group_name(link_text);
+
+        if self.loaded_groups.contains(&group) {
```

Comment:
You can use the return value of `insert` to avoid the map lookup, but I guess that that would make you copy the key so this is probably about as fast (so nvm)